typescript/agentkit/src/action-providers/jupiter/jupiterActionProvider.ts

```diff
@@ -0,0 +1,107 @@
+import { ActionProvider } from "../actionProvider";
+import { Network } from "../../network";
+import { SvmWalletProvider } from "../../wallet-providers/svmWalletProvider";
+import { z } from "zod";
+import { CreateAction } from "../actionDecorator";
+import { SwapTokenSchema } from "./schemas";
+import { PublicKey, VersionedTransaction } from "@solana/web3.js";
+import { createJupiterApiClient, SwapRequest } from "@jup-ag/api";
+
+/**
+ * JupiterActionProvider handles token swaps using Jupiter's API.
+ */
+export class JupiterActionProvider extends ActionProvider<SvmWalletProvider> {
+  /**
+   * Initializes Jupiter API client.
+   */
+  constructor() {
+    super("jupiter", []);
+  }
+
+  /**
+   * Swaps tokens using Jupiter's API.
+   *
+   * @param walletProvider - The wallet provider to use for the swap
+   * @param args - Swap parameters including input token, output token, and amount
+   * @returns A message indicating success or failure with transaction details
+   */
+  @CreateAction({
+    name: "swap",
+    description: `
+    Swaps tokens using Jupiter's DEX aggregator.
+    - Input and output tokens must be valid SPL token mints.
+    - Ensures sufficient balance before executing swap.
+    - Uses Jupiter Blink when available.
```

Comment:
Where does this happen?

Reply:
The old API returned the BLINK link when you requested a quote for a trade. New API does not.

However, apparently building a Jupiter blink ourselves is as simple as building the string `https://jup.ag/swap/${INPUT_TOKEN_SYMBOL}-${OUTPUT_TOKEN_SYMBOL}`. We can easily add that as another tool if the goal is to create blinks. If we want a blink returned, we could compute it in our swap.

Reply:
Going from mint address to symbol is hard and requires using a 3rd party, since symbol data is not stored on the blockchain. It would be better to have the user provide two token names, and the tool just aggregates them together with the base URL if we want users to be able to generate blinks